clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -500,7 +508,19 @@ public Record next() {
                 throw new NoSuchElementException();
 
             readRecords++;
-            return readNext(baseOffset, baseTimestamp, baseSequence, logAppendTime);
+            Record rec = readNext(baseOffset, baseTimestamp, baseSequence, logAppendTime);
+            if (readRecords == numRecords) {
+                // validate the batch
+                boolean isValid = false;
+                try {
+                    readNext(baseOffset, baseTimestamp, baseSequence, logAppendTime);
```

Comment:
Hmm... I will try approaching this via a `read()` call. Let me see if this is practicable.